torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -183,6 +183,41 @@ void FusionExecutor::compileFusion(
     TORCH_INTERNAL_ASSERT(
         out->getValType() == ValType::TensorView,
         "Output types from fusions that are not tensors are not supported at this point.");
+
+    const auto maybe_rfactor_domain =
+        out->as<TensorView>()->getMaybeRFactorDomain();
+    // walking through outputs to see if output shapes are dependent on
+    // non-tensor inputs. For which case, we should have disabled output
+    // allocation, since the caching id only looks at tensor shapes.
+    // See issue https://github.com/csarofeen/pytorch/issues/2002
+    std::vector<Val*> dependencies;
+    for (const auto id : maybe_rfactor_domain) {
+      Val* extent = nullptr;
+      if (id->isReduction() || id->isStride()) {
+        continue;
+      } else if (id->isBroadcast() && id->hasExpandedExtent()) {
+        extent = id->expandedExtent();
+      } else {
+        extent = id->extent();
+      }
+      dependencies.emplace_back(extent);
+    }
+    while (!dependencies.empty()) {
+      if (std::any_of(
+          dependencies.begin(),
+          dependencies.end(),
+          [](Val* val) { return val->isFusionInput(); })) {
+        disable_parameter_cache_ = true;
```

Comment:
Instead of disabling parameter cache completely, does it make sense to add the arguments which are used to calculate the tensor shape into cache id?

Reply:
I think it makes sense to separate the cache logic of the two. Not sure if output sizes would affect launch params. Should double check that.

Meanwhile, I'm sorta trying to turn on cache sooner rather than later :)
I put a TODO there at least for now.

Reply:
Output sizes mean shapes of output tensors? If so, yes they do affect how we calculate threading dimensions.